modules/Media.js

```diff
@@ -2,72 +2,70 @@ import React from 'react'
 import PropTypes from 'prop-types'
 import json2mq from 'json2mq'
 
+const queryType = PropTypes.oneOfType([
+  PropTypes.string,
+  PropTypes.object,
+  PropTypes.arrayOf(PropTypes.object.isRequired)
+])
+
 /**
  * Conditionally renders based on whether or not a media query matches.
  */
 class Media extends React.Component {
   static propTypes = {
-    defaultMatches: PropTypes.bool,
-    query: PropTypes.oneOfType([
-      PropTypes.string,
-      PropTypes.object,
-      PropTypes.arrayOf(PropTypes.object.isRequired)
-    ]).isRequired,
+    defaultMatches: PropTypes.objectOf(PropTypes.bool),
+    queries: PropTypes.objectOf(queryType).isRequired,
     render: PropTypes.func,
-    children: PropTypes.oneOfType([
-      PropTypes.node,
-      PropTypes.func
-    ])
+    children: PropTypes.func
   }
 
-  static defaultProps = {
-    defaultMatches: true
-  }
+  queries = []
 
   state = {
-    matches: this.props.defaultMatches
+    matches:
+      this.props.defaultMatches ||
+      Object.keys(this.props.queries).reduce((acc, key) => ({ ...acc, [key]: true }), {})
   }
 
-  updateMatches = () =>
-    this.setState({ matches: this.mediaQueryList.matches })
+  updateMatches = () => {
+    const newMatches = this.queries.reduce(
+      (acc, { name, mqList }) => ({ ...acc, [name]: mqList.matches }), {}
+    )
+
+    this.setState({ matches: newMatches })
+  }
 
   componentWillMount() {
     if (typeof window !== 'object')
       return
 
-    let { query } = this.props
+    const { queries } = this.props
 
-    if (typeof query !== 'string')
-      query = json2mq(query)
+    this.queries = Object.keys(queries).map(name => {
+      const query = queries[name]
+      const qs = typeof query !== 'string' ? json2mq(query) : query
+      const mqList = window.matchMedia(qs)
+
+      mqList.addListener(this.updateMatches)
+
+      return { name, qs, mqList }
+    })
 
-    this.mediaQueryList = window.matchMedia(query)
-    this.mediaQueryList.addListener(this.updateMatches)
     this.updateMatches()
   }
 
   componentWillUnmount() {
-    this.mediaQueryList.removeListener(this.updateMatches)
+    this.queries.forEach(({ mqList }) =>
+      mqList.removeListener(this.updateMatches)
+    )
   }
 
   render() {
     const { children, render } = this.props
     const { matches } = this.state
 
-    return (
```

Comment:
Let's make changes to the rendering logic in a separate PR, please. As written, this change eliminates support for non-function `children`.

Reply:
Sure! I think I just didn't understand correctly what you meant in the comment :)